test/e2e/storage/non_csi_volumes.go

```diff
@@ -0,0 +1,1567 @@
+		parameters["fsType"] = fsType
+	}
+	ns := c.driverInfo.f.Namespace.Name
+	suffix := fmt.Sprintf("%s-sc", c.driverInfo.name)
+
+	return getStorageClass(provisioner, parameters, ns, suffix)
+}
+
+func (c *cinderDriver) createDriver() {
+	f := c.driverInfo.f
+	ns := f.Namespace
+
+	// We assume that namespace.Name is a random string
+	c.volumeName = ns.Name
+	By("creating a test Cinder volume")
+	output, err := exec.Command("cinder", "create", "--display-name="+c.volumeName, "1").CombinedOutput()
+	outputString := string(output[:])
+	framework.Logf("cinder output:\n%s", outputString)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Parse 'id'' from stdout. Expected format:
+	// |     attachments     |                  []                  |
+	// |  availability_zone  |                 nova                 |
+	// ...
+	// |          id         | 1d6ff08f-5d1c-41a4-ad72-4ef872cae685 |
+	c.volumeID = ""
+	for _, line := range strings.Split(outputString, "\n") {
+		fields := strings.Fields(line)
+		if len(fields) != 5 {
+			continue
+		}
+		if fields[1] != "id" {
+			continue
+		}
+		c.volumeID = fields[3]
+		break
+	}
+	framework.Logf("Volume ID: %s", c.volumeID)
+	Expect(c.volumeID).NotTo(Equal(""))
+}
+
+func (c *cinderDriver) cleanupDriver() {
+	DeleteCinderVolume(c.volumeName)
+}
+
+func DeleteCinderVolume(name string) error {
+	// Try to delete the volume for several seconds - it takes
+	// a while for the plugin to detach it.
+	var output []byte
+	var err error
+	timeout := time.Second * 120
+
+	framework.Logf("Waiting up to %v for removal of cinder volume %s", timeout, name)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(5 * time.Second) {
+		output, err = exec.Command("cinder", "delete", name).CombinedOutput()
+		if err == nil {
+			framework.Logf("Cinder volume %s deleted", name)
+			return nil
+		} else {
+			framework.Logf("Failed to delete volume %s: %v", name, err)
+		}
+	}
+	framework.Logf("Giving up deleting volume %s: %v\n%s", name, err, string(output[:]))
+	return err
+}
+
+// GCE
+type gceDriver struct {
+	volumeName string
+
+	driverInfo driverInfo
+}
+
+func initGceDriver(f *framework.Framework, config framework.VolumeTestConfig, node v1.Node) testDriver {
+	framework.SkipUnlessProviderIs("gce", "gke")
+
+	ns := f.Namespace
+	fsGroup := int64(1234)
+
+	// PD will be created in framework.TestContext.CloudConfig.Zone zone,
+	// so pods should be also scheduled there.
+	config.NodeSelector = map[string]string{
+		kubeletapis.LabelZoneFailureDomain: framework.TestContext.CloudConfig.Zone,
+	}
+
+	return &gceDriver{
+		driverInfo: driverInfo{
+			name:    "gce",
+			fsGroup: &fsGroup,
+			podSec: v1.PodSecurityContext{
+				FSGroup: &fsGroup,
+			},
+			privilegedSecurityContext: true,
+			maxFileSize:               fileSizeMedium,
+			supportedFsType: map[string]bool{
+				"":     true, // Default fsType
+				"ext2": true,
+				"ext3": true,
+				"ext4": true,
+				"xfs":  true,
+			},
+			isBlockSupported: false,
```

Comment:
it is supported

Reply:
Fixed for gce. Also fixed for other drivers.